src/components/BookList.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'; // eslint-disable-line no-unused-vars
+import Book from './Book'; // eslint-disable-line no-unused-vars
+import CategoryFilter from './CategoryFilter' // eslint-disable-line no-unused-vars
+import { REMOVE_BOOK, CHANGE_FILTER } from '../actions'
+import { connect } from 'react-redux'
+
+const BookList = (props) => {
+  const { books } = props
+  const filteredBooks = books.filter((book) => {
+    if (props.filter === '') {
+      return true
+    } else {
+      return props.filter === book.categorie
+    }
+  })
```

Comment:
Hi, wonderful attempt. but we are getting all the book and we don't need that. instead we need to filter the book at the state level and map only the filtered books into our component.
- remember in a previous review i told you about have the state in the component instead of App.

- Now you need to have a function that will read, the state if the filter state in the store and that function will do the filter, and then its the result that should be mapped into the component. like it is now we are getting data we don't need.